### PR TITLE
fix return_one_result flag not return one result

### PR DESCRIPTION
**Pull request by eemjwu**
## Summary

fix return_one_result flag not return one result

## Solution Description

When returning the package, merge the results of each partition.

### CODE REVIEW

File: src/main/java/com/alipay/oceanbase/rpc/table/ObTableClientBatchOpsImpl.java [hunk 1]
```diff
@@ -406,38 +406,50 @@ public void partitionExecute(ObTableOperationResult[] results,
...
 
-        if (subObTableOperationResults.size() < subOperations.getTableOperations().size()) {
-            // only one result when it across failed
-            // only one result when hkv puts
-            if (subObTableOperationResults.size() == 1) {
-                ObTableOperationResult subObTableOperationResult = subObTableOperationResults
+        if (returnOneResult) {
+            if (results[0] == null) {
+                results[0] = new ObTableOperationResult();
+            }
```
Comment (shenyunlong) [inline src/main/java/com/alipay/oceanbase/rpc/table/ObTableClientBatchOpsImpl.java:412]: need to accumulate affected_rows when the table operation spans two or more tablet

File: src/main/java/com/alipay/oceanbase/rpc/table/ObTableClientBatchOpsImpl.java [hunk 2]
```diff
@@ -455,18 +467,35 @@ public ObTableBatchOperationResult executeInternal() throws Exception {
...
-        long start = System.currentTimeMillis();
         List<ObTableOperation> operations = batchOperation.getTableOperations();
-        final ObTableOperationResult[] obTableOperationResults = new ObTableOperationResult[operations
-            .size()];
+        if (operations.isEmpty()) {
+            throw new IllegalArgumentException("operations is empty");
+        }
+        ObTableOperationType lastType = operations.get(0).getOperationType();
+        if (returnOneResult
+                && !(batchOperation.isSameType() && (lastType == ObTableOperationType.INSERT
```
Comment (shenyunlong) [inline src/main/java/com/alipay/oceanbase/rpc/table/ObTableClientBatchOpsImpl.java:476]: can we move this defensive code to the interface layer, which will be more readable?

File: src/main/java/com/alipay/oceanbase/rpc/table/ObTableClientBatchOpsImpl.java (hunk 1 above)
Comment (eemjwu) [reply src/main/java/com/alipay/oceanbase/rpc/table/ObTableClientBatchOpsImpl.java:418]: done

File: src/main/java/com/alipay/oceanbase/rpc/table/ObTableClientBatchOpsImpl.java (hunk 2 above)
Comment (eemjwu) [reply src/main/java/com/alipay/oceanbase/rpc/table/ObTableClientBatchOpsImpl.java]: done

**Review by shenyunlong — APPROVED** [review]

**Pull request merged**